hs_core/migrations/0024_custom_migration_metadata_namespace_20160527.py

```diff
@@ -0,0 +1,34 @@
+from __future__ import unicode_literals
+
+import logging
+
+from django.db import migrations
+
+from hs_core.models import BaseResource
+from hs_core.hydroshare.utils import resource_modified
+
+def migrate_namespace_for_source_and_relation(apps, schema_editor):
+    # migrate the namespace for the 'Source' and 'Relation' metadata
+    # elements from 'dcterms' to 'hsterms' by regenerating the metadata xml
+    log = logging.getLogger()
+    for res in BaseResource.objects.all():
+        # need to regenerate xml file only for those resources that have either 'source' or 'relation' metadata
+        if len(res.metadata.sources.all()) > 0 or len(res.metadata.relations.all()) > 0:
+            resource_modified(res, res.creator)
+            log_msg = "Namespace for either 'source' or 'relation' metadata element was updated for resource " \
```

Comment:
@hyi I think this custom migration is bit different than the one that @gantian127 has for the raster resource. I don't see any problem if the xml file is regenerated more than once by this custom migration. In fact, it is better that the xml file be regenerated each time the related table schema being migrated.